.github/workflows/build.yaml

```diff
@@ -0,0 +1,478 @@
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - uses: ./.github/actions/job-preamble
+
+      - uses: actions/download-artifact@v3
+        with:
+          name: scanner-bundle
+
+      - name: Unpack bundle
+        run: |
+          tar xvzf bundle.tgz
+
+      - uses: actions/download-artifact@v3
+        with:
+          name: scanner-db-bundle
+          path: image/db/rhel
+
+      - name: Build scanner image
+        run: |
+          docker build -t scanner:"$(make --quiet tag)" -f image/scanner/rhel/Dockerfile image/scanner/rhel
+
+      - name: Build scanner-slim image
+        run: |
+            docker build -t scanner-slim:"$(make --quiet tag)" -f image/scanner/rhel/Dockerfile.slim image/scanner/rhel
+
+      - name: Build scanner-db image
+        run: |
+          docker build -t scanner-db:"$(make --quiet tag)" -f image/db/rhel/Dockerfile image/db/rhel
+
+      - name: Build scanner-db-slim image
+        run: |
+          docker build -t scanner-db-slim:"$(make --quiet tag)" -f image/db/rhel/Dockerfile.slim image/db/rhel
+
+        # needed for docs ensure_image.sh initial pull with RHACS_BRANDING
+      - name: Docker login
+        # Skip for external contributions.
+        if: |
+          github.event_name == 'push' || !github.event.pull_request.head.repo.fork
+        run: |
+          docker login -u "${QUAY_RHACS_ENG_RO_USERNAME}" --password-stdin quay.io <<<"${QUAY_RHACS_ENG_RO_PASSWORD}"
+
+      - name: Push images
+        # Skip for external contributions.
+        if: |
+          github.event_name == 'push' || !github.event.pull_request.head.repo.fork
+        run: |
+          source ./scripts/ci/lib.sh
+          push_image_set
+
+  diff-dumps:
+    env:
+      GOOGLE_SA_STACKROX_HUB_VULN_DUMP_UPLOADER: ${{ secrets.GOOGLE_SA_STACKROX_HUB_VULN_DUMP_UPLOADER }}
+      SCANNER_GCP_SERVICE_ACCOUNT_CREDS: ${{ secrets.SCANNER_GCP_SERVICE_ACCOUNT_CREDS }}
+      GOOGLE_SA_CIRCLECI_SCANNER: ${{ secrets.GOOGLE_SA_CIRCLECI_SCANNER }}
+    runs-on: ubuntu-latest
+    needs:
+      - generate-genesis-dump
+    container:
+      image: quay.io/stackrox-io/apollo-ci:scanner-test-0.3.61
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - uses: ./.github/actions/job-preamble
+
+      - uses: actions/download-artifact@v3
+        with:
+          name: updater-build
+
+      - name: Unpack updater build
+        run: |
+          tar xvzf updater-build.tgz
+
+      - uses: actions/download-artifact@v3
+        with:
+          name: genesis-dump
+          path: /tmp/genesis-dump
+
+      - name: diff-dumps
+        run: ./scripts/ci/jobs/diff-dumps.sh
+
+  # TODO(DO NOT MERGE): These store functions don't seem quite right
```

Comment:
what's wrong with them?

Reply:
Basically what Gavin brought up here: https://github.com/stackrox/scanner/pull/1347/files#r1433028812
I'm not sure if these store functions are actually needed in GHA

Reply:
They should be removed. The artifacts are stored earlier (and available through the GH UI) via upload-artifact.